### PR TITLE
Fix Flow and ESLint errors.

### PR DESCRIPTION
**Pull request by tom-un**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and Microsoft/react-native :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into Microsoft/react-native :thumbsup:
- [X] I am making a change required for Microsoft usage of react-native
- [X] I am making a fix / change for the macOS implementation of react-native

## ESLint

There were a large number of ESLint errors due to changes made for macos and windows.
Most of the changes were made the tool itself via:
`npm run lint -- --quiet --fix`
the `--quiet` switch means to silence warning and therefore report and fix only errors, not warnings.

Most of the fixed errors were merely style changes, but three code errors were found and fixed manually:
`accessiblityHint` was added by Microsoft and then later by Facebook.  Most of it was already merged, but in ViewPropTypes the PropType was being exported twice.
DatePickerMacOS.android.js defined `dummyDatePickerMacOS` instead of `DummyDatePickerMacOS`.

in scripts/autoSync.js eslint was complaining about Buffer not being required even though it is a global thing in node.js.   Just added `// eslint-disable-line no-undef`

## Flow
A few Flow issues have crept in since I made a previous change fixing all the flow errors.

ActivityIndicatory.js
color type was change from `string` to `string | SemanticOrDynamicColorType

TouchableWithoutFeedback.js:
`tabIndex: PropTypes.number` and `tooltip: PropTypes.string` needed to be added to the PropTypes

ReactNativeViewAttributes.js
`tabIndex:true` needed to be added

ViewPropTypes.js
`tabIndex?: number` and `tabIndex: PropTypes.number` added

There are a handful of errors in node_modules file not directly in our control.   I just exclude them via the .flowconfig file.  The rn-cli.config.js has some SDX platform things added that don't work when not a subtree of sdx-platform so I also excluded that file.

/azp run

### CODE REVIEW

File: .flowconfig [hunk 1]
```diff
@@ -23,7 +23,11 @@
 .*/Libraries/polyfills/.*
 
 ; Ignore metro
-; .*/node_modules/metro/.*
+.*/node_modules/metro/.*
```
Comment (HeyImChris) [inline .flowconfig]: How close does this get us to the CI loops passing? If we're still not close, there's always the risk these just break again without people realizing.

Comment (HeyImChris) [inline .flowconfig]: Is sdx-platform locked for RN changes yet? If not, we could get into a lot of painful merge conflicts next time we merge this fork into it

Comment (acoates-ms) [reply .flowconfig]: You need to bypass various checks to get it into sdx-platform these days.  We should be good here.

**Pull request merged**